key_value_store.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <iostream>
+#include <unordered_map>
+#include <mutex>
+#include <boost/functional/hash.hpp>
```

Comment:
Some people like to order the headers by either library (like std or boost) or alphabetically, or by right jagged-ness. I tend to like to separate the library headers by newlines:
```c++
#include <mutex>
#include <iostream>
#include <unordered_map>

#include <boost/functional/hash.hpp>
```